src/frontend/src/components/forms/ApiForm.tsx

```diff
@@ -222,41 +224,46 @@ export function ApiForm({ id, props }: { id: string; props: ApiFormProps }) {
       props.pathParams
     ],
     queryFn: async () => {
-      return api
-        .get(url)
-        .then((response) => {
-          const processFields = (fields: ApiFormFieldSet, data: NestedDict) => {
-            const res: NestedDict = {};
-
-            for (const [k, field] of Object.entries(fields)) {
-              const dataValue = data[k];
-
-              if (
-                field.field_type === 'nested object' &&
-                field.children &&
-                typeof dataValue === 'object'
-              ) {
-                res[k] = processFields(field.children, dataValue);
-              } else {
-                res[k] = dataValue;
-              }
+      try {
+        // Await API call
+        let response = await api.get(url);
+        // Define function to process API response
+        const processFields = (fields: ApiFormFieldSet, data: NestedDict) => {
+          const res: NestedDict = {};
+
+          // TODO: replace with .map()
```

Comment:
Question (non blocking): can this be addressed now?

Reply:
Don't think I understand the logic of that loop well enough to refactor that properly yet